src/main/java/edu/cornell/kfs/coa/document/validation/impl/AccountGlobalPreRules.java

```diff
@@ -0,0 +1,117 @@
+            	accountGlobal.setEndowmentIncomeAcctFinCoaCd(account.getChartOfAccountsCode());
+            }
+        }
+
+        if (StringUtils.isNotBlank(accountGlobal.getIncomeStreamAccountNumber())) {
+            Account account = checkForContinuationAccount("Income Stream Account", accountGlobal.getIncomeStreamFinancialCoaCode(), accountGlobal.getIncomeStreamAccountNumber(), "");
+            if (ObjectUtils.isNotNull(account)) { // override old user inputs
+            	accountGlobal.setIncomeStreamAccountNumber(account.getAccountNumber());
+            	accountGlobal.setIncomeStreamFinancialCoaCode(account.getChartOfAccountsCode());
+            }
+        }
+
+        if (StringUtils.isNotBlank(accountGlobal.getContractControlAccountNumber())) {
+            Account account = checkForContinuationAccount("Contract Control Account", accountGlobal.getContractControlFinCoaCode(), accountGlobal.getContractControlAccountNumber(), "");
+            if (ObjectUtils.isNotNull(account)) { // override old user inputs
+            	accountGlobal.setContractControlAccountNumber(account.getAccountNumber());
+            	accountGlobal.setContractControlFinCoaCode(account.getChartOfAccountsCode());
+            }
+        }
+
+        for (IndirectCostRecoveryAccountChange icra : accountGlobal.getActiveIndirectCostRecoveryAccounts()){
+            if (StringUtils.isNotBlank(icra.getIndirectCostRecoveryAccountNumber())) {
+                Account account = checkForContinuationAccount("Indirect Cost Recovery Account", icra.getIndirectCostRecoveryAccountNumber(), icra.getIndirectCostRecoveryFinCoaCode(), "");
+                if (ObjectUtils.isNotNull(account)) { // override old user inputs
+                    icra.setIndirectCostRecoveryAccountNumber(account.getAccountNumber());
+                    icra.setIndirectCostRecoveryFinCoaCode(account.getChartOfAccountsCode());
+                }
+            }
+        }
+    }
+    
+    protected void checkForDefaultSubFundGroupStatus() {
+        String restrictedStatusCode = StringUtils.EMPTY;
+
+        if (ObjectUtils.isNull(accountGlobal.getSubFundGroup()) || StringUtils.isBlank(accountGlobal.getSubFundGroupCode())) {
+            return;
+        }
+        SubFundGroup subFundGroup = accountGlobal.getSubFundGroup();
+
+        if (StringUtils.isNotBlank(subFundGroup.getAccountRestrictedStatusCode())) {
+            restrictedStatusCode = subFundGroup.getAccountRestrictedStatusCode().trim();
+            accountGlobal.setAccountRestrictedStatusCode(restrictedStatusCode);
+        }
+
+    }
+    
+    @SuppressWarnings("deprecation")
+    protected boolean checkOffCampus() {
+      boolean continueRules = true;
+
+      if (accountGlobal.getAccountOffCampusIndicator() !=null && accountGlobal.getAccountOffCampusIndicator()) {
+        String questionText = SpringContext.getBean(ConfigurationService.class).getPropertyValueAsString(CUKFSKeyConstants.QUESTION_ACCOUNT_OFF_CAMPUS_INDICATOR);
```

Comment:
Another more SpringContext.getBean() call that could be replaced by a getter/setter.